File: meetup.py
import requests
from datetime import datetime, date, time
output = open('meetup.html', 'w+', encoding='utf-8')
output.write('<!DOCTYPE html><html><head></head><body><h1>EVENTS</h1>')
response = requests.get('https://api.meetup.com/2/open_events?&sign=true&photo-host=public&country=ca&topic=newtech&city=Montreal&key=1110b4476750362f751e3024402a13').json()
for i in range(7):
    if i == 0: output.write('<h2>Monday</h2>')
    elif i == 1: output.write('<h2>Tuesday</h2>')
    elif i == 2: output.write('<h2>Wednesday</h2>')
    elif i == 3: output.write('<h2>Thursday</h2>')
    elif i == 4: output.write('<h2>Friday</h2>')
    elif i == 5: output.write('<h2>Saturday</h2>')
    elif i == 6: output.write('<h2>Sunday</h2>')
    for item in response['results']:
        d = datetime.fromtimestamp(int(item['time']) / 1000).weekday()
        if d == i:
            try:
                output.write('<div><p>Time: ' + str(datetime.fromtimestamp(int(item['time']) / 1000)) + '</p>')
            except KeyError:pass
            try:
                output.write('<p>Name: ' + str(item['name'] + '</p>'))
            except KeyError:pass
            try:
                output.write('<p>Organizer: ' + str((item['group'])['name'])+'</p>')
            except KeyError:pass
            try:
                output.write('<p>Description: ' + str(item['description']) + '</p>')
            except KeyError:pass
            try:
                output.write("<p>Phone: " + str((item['venue'])['phone']) + '</p>')
            except KeyError:pass
            try:
                output.write("<p>Country: " + str((item['venue'])['localized_country_name']) + '</p>')
            except KeyError:pass
            try:
                output.write('<p>City: ' + str((item['venue'])['city']) + '</p>')
            except KeyError:pass
            try:
                output.write('<p>Address: ' + str((item['venue'])['address_1']) + '</p>')
            except KeyError:pass
            output.write('</div>')
output.write('</body></html>')
output.close()

File: meetup.rb
require 'meetup_client'

MeetupClient.configure do |config|
  config.api_key = '1110b4476750362f751e3024402a13'
end

params = { country: 'ca',
	city: 'Montreal',
	topic: 'newtech',
	status: 'upcoming',
    format: 'json' }
meetup_api = MeetupApi.new
events = meetup_api.open_events(params)

output = File.new('meetup.html', 'w')
output.puts('<!DOCTYPE html><html><head></head><body><h1>EVENTS</h1>')
for i in 0..6
	case i
	when 0 
		output.puts('<h2>Sunday</h2>')
	when 1 
		output.puts('<h2>Monday</h2>')
	when 2 
		output.puts('<h2>Tuesday</h2>')
	when 3 
		output.puts('<h2>Wednesday</h2>')
	when 4 
		output.puts('<h2>Thursday</h2>')
	when 5 
		output.puts('<h2>Friday</h2>')
	when 6 
		output.puts('<h2>Saturday</h2>')
	end
	for result in events['results']
		day = Time.at(result['time'] / 1000).wday
		if (day == i)
			output.puts('<div><p>Time: ' + result['time'].to_s + '</p>')
			output.puts('<p>Name: ' + result['name'].to_s + '</p>')
			output.puts('<p>Address: ' + result['name']['address_1'].to_s + '</p>')
			output.puts('<p>Description: ' + result['description'].to_s + '</p>')
			output.puts('</div>')
		end
	end
end
output.puts('</body></html>')
output.close

puts 'meetup.html created'
